spark3-extensions/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ExtendedDataSourceV2Strategy.scala

```diff
@@ -20,14 +20,25 @@
 package org.apache.spark.sql.execution.datasources.v2
 
 import org.apache.spark.sql.{AnalysisException, Strategy}
-import org.apache.spark.sql.catalyst.plans.logical.{CallStatement, LogicalPlan}
-import org.apache.spark.sql.execution.SparkPlan
+import org.apache.spark.sql.catalyst.plans.logical.{CallStatement, DynamicFileFilter, LogicalPlan, OverwriteFiles}
+import org.apache.spark.sql.execution.{ProjectExec, SparkPlan}
 
 object ExtendedDataSourceV2Strategy extends Strategy {
 
   override def apply(plan: LogicalPlan): Seq[SparkPlan] = plan match {
     case _: CallStatement =>
       throw new AnalysisException("CALL statements are not currently supported")
+    case DynamicFileFilter(scanRelation, fileFilterPlan) =>
+      val scanExec = ExtendedBatchScanExec(scanRelation.output, scanRelation.scan)
```

Comment:
I think you should mention that this doesn't use `planLater` because we need to use `ExtendedBatchScanExec` instead of `BatchScanExec`.